### DISCUSSION TITLE
Download große Region komplett / Download large region completely

### DISCUSSION THREAD

**Discussion by ToToGSARider** [devemux86/cruiser#429] [Q&A] [answered]
Guten Morgen, ich probiere aktuell die Karte Frankreich komplett zu laden mit 2,8 GB. Das Handy bzw. Cruiser sagte mir, dass der Download abgeschlossen ist, aber die Karten waren nicht zu sehen. Ich habe den Download jetzt noch einmal angestossen. Werden die bereits geladenen Karten dann ggf. ersetzt, falls sich der Download verschluckt hat. Wie komme ich in einem Samsung Gerät an den Speicherordner? er ist in den Dateien nicht sichtbar. Bei Deutschland hatte ich ja dieses Problem auch schon einmal, dass man 2-3 mal starten mußte.

Good morning, I am currently trying to load the full 2.8 GB map of France. The cell phone or Cruiser told me that the download was complete, but the maps were not visible. I have now started the download again. Will the maps that have already been downloaded be replaced if the download is interrupted? How do I get to the storage folder in a Samsung device? It is not visible in the files. I have had this problem with Germany before, where you had to start 2-3 times.

**Comment by devemux86** [comment, marked as answer]
@ToToGSARider 

To download large maps, it is better to leave the screen on and make sure that Cruiser is not running in the background so that Android does not terminate it.
The storage folder can be selected in "Settings | Application | Advanced".

Um große Karten herunterzuladen, lassen Sie den Bildschirm besser eingeschaltet und stellen Sie sicher, dass Cruiser nicht im Hintergrund läuft, damit Android es nicht beendet.
Der Speicherordner kann unter "Einstellungen | Applikation | Erweitert" ausgewählt werden.

**Reply by ToToGSARider** [reply]
The download has now worked. Could you also tell Cruiser to open the storage folder? Does Cruiser recognize whether files already exist during the download and update them if the download fails?

Der Download hat jetzt funktioniert. Könnte man in Cruiser auch sagen, öffne den Speicherordner? Erkennt Cruiser beim Download, ob Dateien schon vorhanden sind und aktualisiert sie ggf. bei fehlgeschlagenen Downloads?

**Reply by devemux86** [reply]
@ToToGSARider 

Downloads must be successfully completed while Cruiser is running in order to be processed.
Each server provides downloads in different formats, so processing is required after downloading.

Die Downloads müssen erfolgreich abgeschlossen werden, während Cruiser läuft, damit sie verarbeitet werden können.
Jeder Server stellt Downloads in unterschiedlichen Formaten zur Verfügung, so dass nach dem Herunterladen eine Bearbeitung erforderlich ist.

**Comment by ToToGSARider** [comment]
> Der Speicherordner kann unter "Einstellungen | Applikation | Erweitert" ausgewählt werden.

Könnte man hier einbauen, öffne Speicherort? 

Could you include open location here?

**Reply by 0709wiwiwi** [reply]
_> The storage folder can be selected in "Settings | Application | Advanced"._
![image](https://github.com/devemux86/cruiser/assets/54167201/95377a64-91a9-4083-90c6-0d76f8b72d6d)
File explorer accesses in A13+.
Android/data is not accessible.
Android/media is accessible.
Offers additional options for devs?
Post sensitive items in Android/data.
Post public items in Android/media.

**Reply by devemux86** [reply]
@ToToGSARider 

The storage locations are defined by Android restrictions.
You can select the media folder (see image above) to use it with a file manager.

Die Speicherorte werden durch Android-Einschränkungen definiert.
Sie können den Medienordner auswählen (siehe Bild oben), um ihn mit einem Dateimanager zu verwenden.

**Comment by ToToGSARider** [comment]
> @ToToGSARider
> 
> The storage locations are defined by Android restrictions. You can select the media folder (see image above) to use it with a file manager.
> 
> Die Speicherorte werden durch Android-Einschränkungen definiert. Sie können den Medienordner auswählen (siehe Bild oben), um ihn mit einem Dateimanager zu verwenden.

Okay, sollte ich dann besser erst einmal alle Karten löschen oder werden die Karten dann rüber geschoben?

Die Frage ist, wie schon erwähnt, erkennt der Cruiser bereits vorhandene Karten wenn der Download nicht erfolgreich abgeschlossen wurde und aktualisiert er diese. Werden die Karten überhaupt gespeichert, wenn der Download nicht erfolgreich war? 

Okay, should I delete all the cards first or will the cards then be moved over? 

The question is, as already mentioned, does the Cruiser recognize existing maps if the download was not successful and does it update them? Are the maps saved at all if the download was not successful?

**Reply by devemux86** [reply]
@ToToGSARider 

> Okay, sollte ich dann besser erst einmal alle Karten löschen oder werden die Karten dann rüber geschoben?

When the storage folder is changed, the files in the old folder are automatically deleted and must be downloaded again.

Beim Wechsel des Speicherordners werden die Dateien im alten Ordner automatisch gelöscht und müssen neu heruntergeladen werden.

> Die Frage ist, wie schon erwähnt, erkennt der Cruiser bereits vorhandene Karten wenn der Download nicht erfolgreich abgeschlossen wurde und aktualisiert er diese. Werden die Karten überhaupt gespeichert, wenn der Download nicht erfolgreich war?

Downloads are performed by Android and not by Cruiser. Cruiser takes the successful downloads and displays them in the app.

Downloads werden von Android und nicht von Cruiser durchgeführt. Cruiser nimmt die erfolgreichen Downloads und zeigt sie in der App an.

**Comment by devemux86** [comment]
~Next Cruiser update will use the public download folder so you can watch and clear it if you like:~
~`/storage/emulated/0/Download/`~

This is for the downloads made by Android. The final storage location remains the selected storage folder.

You should test with your devices and Android versions to make sure everything is working properly.

**Reply by 0709wiwiwi** [reply]
KISS? Sounds complicated. With consistent use of android/media, even A13+ is still friendly. 
By  a nice (cx) file explorer easily add shortcuts to any Cruiser folder like your maps and extra.

**Reply by devemux86** [reply]
@0709wiwiwi there are 2 different folders: download folder (invisible to users) and storage folder.

- The **download** folder is only used by Android and is restricted to 2 locations: internal and external.
Now we just try the external location. Android downloads files in this folder, then apps take them.

- The **storage** folder - where downloads are transferred and stored by the apps - remains the same
and is freely selected by the users.

For most users nothing has really changed.

**Reply by ToToGSARider** [reply]
Ich habe jetzt eine Lösung gefunden wie ich die regulären Ordner sehen kann. Wenn das Handy Samsung S20+ mit dem PC verbunden ist, sehe ich alle Ordner. 
Es wäre jetzt interessant zu wissen, wo die angeblich abgeschlossenen Downloads hingekommen sind, die aber im Cruiser dann trotzdem nicht sichtbar waren. Ich hatte beim 1. gelungenen Download die Meldung abgeschlossen, aber die Karte war trotzdem nicht da. Wo ist die dann hingekommen?

I have now found a solution how I can see the regular folders. When the Samsung S20+ phone is connected to the PC, I can see all the folders. 
It would be interesting to know where the supposedly completed downloads went, but they were still not visible in Cruiser. I had the message completed on the 1st successful download, but the card was still not there. Where did it go then?

**Reply by devemux86** [reply]
@ToToGSARider 

Downloads are received by Android in an internal system folder. If they are not used, they are probably removed.

Downloads werden von Android in einem internen Systemordner empfangen. Wenn sie nicht verwendet werden, werden sie wahrscheinlich entfernt.

**Reply by ToToGSARider** [reply]
> Next Cruiser update will use the public download folder so you can watch and clear it if you like:
> `/storage/emulated/0/Download/`
> 
> This is for the downloads made by Android. The final storage location remains the selected storage folder.
> 
> You should test with your devices and Android versions to make sure everything is working properly.

Aber die Karten werden dann aus dem Download Ordner in die App geholt, wenn sie erfolgreich geladen und aktiviert wurden?

But the maps are then fetched from the download folder into the app once they have been successfully downloaded and activated?

**Reply by devemux86** [reply]
@ToToGSARider 

> Aber die Karten werden dann aus dem Download Ordner in die App geholt, wenn sie erfolgreich geladen und aktiviert wurden?

Correct, they are unpacked / moved and then they can be activated.

Richtig, sie werden entpackt / verschoben und dann können sie aktiviert werden.

**Reply by 0709wiwiwi** [reply]
> Wenn das Handy Samsung S20+ mit dem PC verbunden ist, sehe ich alle Ordner.

Lösung ?

**Reply by ToToGSARider** [reply]
Ja, für mich ist das schon Okay. Wenn natürlich der Download Ordner jetzt geändert wird, ist das die perfekte Lösung. Ich bin mit dem Stand jetzt zufrieden. 

Yes, that's okay for me. Of course, if the download folder is changed now, that's the perfect solution. I'm happy with the way things are now.

**Reply by devemux86** [reply]
@ToToGSARider 

We will revert the download folder to the previous location as it does not work on all devices.
Android and companies do not like public folders.

https://github.com/devemux86/cruiser/discussions/414#discussioncomment-8247745

Wir werden den Download-Ordner wieder auf die vorherigen Speicherorte zurücksetzen, da er nicht auf allen Geräten funktioniert.
Android und Unternehmen mögen keine öffentlichen Ordner.

**Reply by 0709wiwiwi** [reply]
@ToToGSARider 
I use cx file explorer.  (Google play)
https://cxfileexplorer.com/how-to-use-cx-file-explorer/

**Must set** : Show advanced menus enables additional functionality of Cx File Explorer:
- Enables **Add to Home Screen button in the advanced menu options.** 
Condition on A13+ :  Cruiser storage folder in Android/media !
Offers fast acces to the Cruiser maps folder. (I had a test on A13)
You need one single tap, no pc necessary.
![20240125_194947](https://github.com/devemux86/cruiser/assets/54167201/69f2c356-3a80-4d6e-a1dc-ecdb6f9428f2)
